src/main/java/chess/dao/SquareDaoImpl.java

```diff
@@ -0,0 +1,62 @@
+package chess.dao;
+
+import chess.entity.SquareEntity;
+import chess.model.board.Board;
+import chess.model.piece.Piece;
+import chess.model.position.Position;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class SquareDaoImpl implements SquareDao {
+
+    private JdbcTemplate jdbcTemplate;
+
+    public SquareDaoImpl(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    private final RowMapper<SquareEntity> actorRowMapper = (resultSet, rowNum) -> {
+        SquareEntity squareEntity = new SquareEntity(
+                resultSet.getString("position"),
+                resultSet.getString("team"),
+                resultSet.getString("symbol")
+        );
+        return squareEntity;
+    };
+
+    public void insert(Position position, Piece piece) {
+        final String sql = "insert into square (position, team, symbol) values (?, ?, ?)";
+        jdbcTemplate.update(sql, position.getKey(), piece.getTeam(), piece.getSymbol());
+    }
+
+    public Board createBoard() {
+        List<SquareEntity> squareEntities = findAll();
+        Map<Position, Piece> squares = new HashMap<>();
+        for (SquareEntity squareEntity : squareEntities) {
+            Position position = Position.from(squareEntity.getPosition());
+            Piece piece = Piece.getPiece(squareEntity.getTeam() + "_" + squareEntity.getSymbol());
```

Comment:
```suggestion
            Piece piece = Piece.find(squareEntity.getTeam(), squareEntity.getSymbol());
```

SquareDaoImple 객체가 Piece 내부를 잘 알고 있어야 메서드를 쓸 수 있네요.
피스의 키 값에 언더바가 필요하다는걸 어떻게 알 수 있을까요?
객체 내부 사정을 몰라도 쓸 수 있게 만들어주세요.

Reply:
키 값에 언더바를 적용하는 것을 Piece 객체에게 책임을 주어 Piece 객체의 내부사정을 몰라도 사용 할 수 있도록 변경하였어요!